debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/ChangeStreamPipelineFactory.java

```diff
@@ -120,39 +126,89 @@ private ChangeStreamPipeline createUserPipeline() {
         return filterConfig.getUserPipeline();
     }
 
-    private static Optional<Bson> createCollectionFilter(FilterConfig filterConfig) {
+    private static Optional<Bson> createDatabaseAndCollectionRegexFilters(FilterConfig filterConfig) {
         // Database filters
         // Note: No need to exclude `filterConfig.getBuiltInDbNames()` since these are not streamed per
         // https://www.mongodb.com/docs/manual/changeStreams/#watch-a-collection--database--or-deployment
-        var dbFilters = Optional.<Bson> empty();
-        if (filterConfig.getDbIncludeList() != null) {
-            dbFilters = Optional.of(Filters.regex("event.ns.db", filterConfig.getDbIncludeList().replaceAll(",", "|"), "i"));
-        }
-        else if (filterConfig.getDbExcludeList() != null) {
-            dbFilters = Optional.of(Filters.regex("event.ns.db", "(?!" + filterConfig.getDbExcludeList().replaceAll(",", "|") + ")", "i"));
-        }
+        var dbFilters = Optional.<Bson> empty()
+                .or(() -> filterConfig.getDbIncludeList()
+                        .map(value -> Filters.regex("event.ns.db", value.replaceAll(",", "|"), "i")))
+                .or(() -> filterConfig.getDbExcludeList()
+                        .map(value -> Filters.regex("event.ns.db", "(?!" + value.replaceAll(",", "|") + ")", "i")));
 
         // Collection filters
-        var collectionsFilters = Optional.<Bson> empty();
-        if (filterConfig.getCollectionIncludeList() != null) {
-            collectionsFilters = Optional
-                    .of(Filters.regex("namespace", filterConfig.getCollectionIncludeList().replaceAll(",", "|"), "i"));
-        }
-        else if (filterConfig.getCollectionExcludeList() != null) {
-            collectionsFilters = Optional
-                    .of(Filters.regex("namespace", "(?!" + filterConfig.getCollectionExcludeList().replaceAll(",", "|") + ")", "i"));
-        }
+        var collectionsFilters = Optional.<Bson> empty()
+                .or(() -> filterConfig.getCollectionIncludeList()
+                        .map(value -> Filters.regex("namespace", value.replaceAll(",", "|"), "i")))
+                .or(() -> filterConfig.getCollectionExcludeList()
+                        .map(value -> Filters.regex("namespace", "(?!" + value.replaceAll(",", "|") + ")", "i")));
+
+        return andFilters(
+                dbFilters,
+                collectionsFilters);
+    }
+
+    private static Optional<Bson> createDatabaseAndCollectionLiteralFilters(FilterConfig filterConfig) {
+        // Database filters
+        // Note: No need to exclude `filterConfig.getBuiltInDbNames()` since these are not streamed per
+        // https://www.mongodb.com/docs/manual/changeStreams/#watch-a-collection--database--or-deployment
+        var dbFilters = Optional.<Bson> empty()
+                .or(() -> filterConfig.getDbIncludeList()
+                        .map(value -> value.split(","))
```

Comment:
Should we trim whitespace for the tokens?

Reply:
☝️ Said formatting

Reply:
mapping to namespace document takes care of that (should be the next line)

Reply:
Hmm, not seeing it. At any rate, can we add a test for this configuration value?

`" db1.A, db1.B ,  db3.C "`